kedro_azureml/runner.py

```diff
@@ -33,25 +46,42 @@ def run(
         hook_manager: PluginManager = None,
         session_id: str = None,
     ) -> Dict[str, Any]:
-        unsatisfied = pipeline.inputs() - set(catalog.list())
-        for ds_name in unsatisfied:
-            catalog = catalog.shallow_copy()
-            catalog.add(ds_name, self.create_default_data_set(ds_name))
+        catalog = catalog.shallow_copy()
+        catalog_set = set(catalog.list())
+
+        # Loop over datasets in arguments to set their paths
+        for ds_name, ds_path in self.data_paths.items():
+            if ds_name in catalog_set:
+                ds = catalog._get_dataset(ds_name)
+                if isinstance(ds, AzureMLFolderDataset):
+                    ds.path = str(Path(ds_path) / Path(ds.path).name)
+                    catalog.add(ds_name, ds, replace=True)
+            else:
```

Comment:
It should, but I can add a test for that

Reply:
Please do 👍

Reply:
Update: this does not work. When `native_data_passing` is set to `false`, my pipeline fails:
```
OSError: [Errno 30] Read-only file system: 
'/mnt/azureml/cr/j/bf93be579501460a81ac6d458beac2ef/cap/data-capability/wd/INPUT
_preprocessed_shuttles/output.txt'
```

It seems like the logic was not preserved by your change.

Reply:
I have fixed this now in the `execute` command. However, `test_runner_fills_missing_datasets` still fails because it doesn't use the CLI arguments (`az-input` and `az-output`) which my logic relies on. Is it OK to replace this test by one that relies on the CLI arguments?